src/main/java/org/apache/sysml/api/python/SystemML.py

```diff
@@ -247,3 +259,161 @@ def getStringRDD(self, varName, format):
         #    return rdd
         #except Py4JJavaError:
         #    traceback.print_exc()
+
+def getNumCols(numPyArr):
+    if numPyArr.ndim == 1:
+        return 1
+    else:
+        return numPyArr.shape[1]
+       
+def convertToMatrixBlock(sc, src):
+    if isinstance(sc, SparkContext):
+        src = np.asarray(src)
+        numCols = getNumCols(src)
+        numRows = src.shape[0]
+        arr = src.ravel().astype(np.float64)
+        buf = bytearray(arr.tostring())
+        return sc._jvm.org.apache.sysml.runtime.instructions.spark.utils.RDDConverterUtilsExt.convertPy4JArrayToMB(buf, numRows, numCols)
+    else:
+        raise TypeError('sc needs to be of type SparkContext') # TODO: We can generalize this by creating py4j gateway ourselves
+    
+
+def convertToNumpyArr(sc, mb):
+    if isinstance(sc, SparkContext):
+        numRows = mb.getNumRows()
+        numCols = mb.getNumColumns()
+        buf = sc._jvm.org.apache.sysml.runtime.instructions.spark.utils.RDDConverterUtilsExt.convertMBtoPy4JDenseArr(mb)
+        return np.frombuffer(buf, count=numRows*numCols, dtype=np.float64)
+    else:
+        raise TypeError('sc needs to be of type SparkContext') # TODO: We can generalize this by creating py4j gateway ourselves
+
+class mllearn:
+    # Or we can create new Python project with package structure
+    class LogisticRegression(Estimator):
+
+        def __init__(self, sqlCtx, penalty='l2', fit_intercept=True, max_iter=100, max_inner_iter=0, tol=0.000001, C=1.0, solver='newton-cg', transferUsingDF=False):
+            self.sqlCtx = sqlCtx
+            self.sc = sqlCtx._sc
+            self.log = self.sc._jvm.org.apache.sysml.api.ml.LogisticRegression("lr", self.sc._jsc.sc())
+            self.transferUsingDF = transferUsingDF
+            if penalty != 'l2':
+                raise Exception('Only l2 penalty is supported')
+            self.icpt = int(fit_intercept)
+            self.max_iter = max_iter
+            self.max_inner_iter = max_inner_iter
+            self.tol = tol
+            if C < 0:
+                raise Exception('C has to be positive')
+            reg = 1.0 / C
+            self.reg = reg
+            self.updateLog()
+            if solver != 'newton-cg':
+                raise Exception('Only newton-cg solver supported')
+             
+        def updateLog(self):
+            self.log.setMaxOuterIter(self.max_iter)
+            self.log.setMaxInnerIter(self.max_inner_iter) 
+            self.log.setRegParam(self.reg)
+            self.log.setTol(self.tol)
+            self.log.setIcpt(self.icpt)
+            
+        def convertToPandasDF(self, X):
+            if isinstance(X, np.ndarray):
+                colNames = []
+                numCols = getNumCols(X)
+                for i in range(0, numCols):
+                    colNames = colNames + [ str('C' + str(i))]
+                pdfX = pd.DataFrame(X, columns=colNames)
```

Comment:
It is safe to say that you can move this method out of this class, nothing in the implementation seems to suggest it belongs to LogisticRegression